nautobot_ssot/jobs/base.py

```diff
@@ -179,7 +181,12 @@ def record_memory_trace(step: str):
         )
         if memory_profiling:
             record_memory_trace("target_load")
-
+        # Check if the adapter is an instance of NautobotAdapter to determine if it's a contrib implementation,
+        # in which case we should create the required MetadataType object.
+        if self.__class__.__name__ in settings.PLUGINS_CONFIG.get("nautobot_ssot", {}).get(
+            "enable_metadata_for", []
+        ) and isinstance(self.target_adapter, NautobotAdapter):
+            self.target_adapter.get_or_create_metadatatype()
```

Comment:
I see now the difficulty in making this a property, maybe you can still make it work but the current approach is still okay for me.